.github/workflows/pythonpackage.yml

```diff
@@ -0,0 +1,96 @@
+name: Tests and lint
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v*"
+  pull_request:
+    branches:
+      - "*"
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Use a matrix for the python version here to make the rest of the lint job setup
+        # as similar as possible to the test job below.
+        python-version: [3.8]
+        lint-flags:
+          - "--run-only-fast-linters"
+          - "--run-only-pylint"
+          - "--run-only-mypy"
+          - "--run-only-bandit"
+          - "--run-only-sphinx-build"
+          - "--run-only-typing-copilot-tighten"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install core dependencies
+        run: |
+          ./scripts/install_ubuntu_ci_core_dependencies.sh
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          echo "::set-output name=dir::$(pip cache dir)"
+      - name: Cache the Python dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/Pipfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-${{ matrix.python-version }}-
```

Comment:
Is this what the GH docs recommend? I'm surprised that there's so much boilerplate, especially to establish the cache directory. The Node cache setup is much terser; totally possible that Python is different but wanted to double check.

Reply:
What does the node cache setup look like? This is based on https://github.com/actions/cache/blob/main/examples.md#using-pip-to-get-cache-location

We don't currently do multi-OS testing, but I didn't feel like debugging the Actions setup if we ever do end up doing it.

Reply:
Ah, I use something like [this](https://github.com/actions/cache/blob/d606e039ae32f64a8593bf4a37b0bf205c695237/examples.md#macos-and-ubuntu) but it only works on *nix and the Windows snippet they recommend in the subsequent section looks more like the pip one. Makes sense.